specification/network/resource-manager/Microsoft.Network/stable/2018-12-01/webapplicationfirewall.json

```diff
@@ -0,0 +1,530 @@
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "maxLength": 128,
+            "description": "The name of the policy."
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted. The request has been accepted for processing and the operation will complete asynchronously."
+          },
+          "204": {
+            "description": "No Content. The request has been accepted but the policy was not found."
+          },
+          "200": {
+            "description": "Delete successful."
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    }
+  },
+  "definitions": {
+    "WebApplicationFirewallPolicy": {
+      "description": "Defines web application firewall policy.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/WebApplicationFirewallPolicyPropertiesFormat",
+          "description": "Properties of the web application firewall policy."
+        },
+        "etag": {
+          "type": "string",
+          "description": "Gets a unique read-only string that changes whenever the resource is updated."
+        }
+      },
+      "allOf": [
+        {
+            "$ref": "./network.json#/definitions/Resource"
+        }
+      ]
+    },
+    "WebApplicationFirewallPolicyPropertiesFormat": {
+      "description": "Defines web application firewall policy properties",
+      "properties": {
+        "policySettings": {
+          "description": "Describes  policySettings for policy",
+          "$ref": "#/definitions/PolicySettings"
+        },
+        "customRules": {
+          "description": "Describes custom rules inside the policy",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CustomRule"
+          }
+        },
+        "applicationGateways": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "./applicationGateway.json#/definitions/ApplicationGateway"
+          },
+          "description": "A collection of references to application gateways."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Provisioning state of the WebApplicationFirewallPolicy."
+        },
+        "resourceState": {
+          "title": "Resource status of the policy.",
+          "readOnly": true,
+          "enum": [
+            "Creating",
+            "Enabling",
+            "Enabled",
+            "Disabling",
+            "Disabled",
+            "Deleting"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "WebApplicationFirewallPolicyPropertiesFormat",
```

Comment:
This enum has the same name it's parent definition has, this could cause collisions in generated SDK. Plus as the name of enum it's not really clear. Please rename it to something like `WebApplicationFirewallPolicyResourceState`

Reply:
made the change